kubehelpers/psm.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 gRPC authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubehelpers
+
+import (
+	"fmt"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+)
+
+// IsPSMTest checks if a given Loadtest is a PSM test. All clients running
+// a scenario should have the same configuration otherwise a error will be
+// returned.
+func IsPSMTest(clients *[]grpcv1.Client) (bool, error) {
```

Comment:
It kind of feels like this function is doing two things: validating the test spec and then seeing what type of test spec it is.

Maybe split this out into two functions? Then you would only need one place in the code asserting that all clients must run the same kind of test.